src/main/fc/fc_core.c

```diff
@@ -238,9 +238,9 @@ static void updateArmingStatus(void)
         /* CHECK: Arming switch */
         if (!isUsingSticksForArming()) {
             // If arming is disabled and the ARM switch is on
-            if (isArmingDisabled() && IS_RC_MODE_ACTIVE(BOXARM)) {
+            if (isArmingDisabled() && !IS_RC_MODE_ACTIVE(BOXARM)) {     //kbi Issue #2120 3D Mode Arming Not Working
```

Comment:
This is incorrect. If arming was disabled before and arming switch is "ON" we need to keep arming disabled.